engine/src/test/java/io/camunda/zeebe/engine/processing/incident/MultiInstanceIncidentTest.java

```diff
@@ -172,6 +175,64 @@ public void shouldCreateIncidentIfOutputElementExpressionEvaluationFailed() {
                 + "'ValError(no variable found for name 'undefined_var')'");
   }
 
+  @Test
+  public void shouldCollectOutputResultsForResolvedIncidentOfOutputElementExpression() {
+    // given an instance of a process with an output mapping referring to a variable 'undefined_var'
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(MULTI_TASK_PROCESS)
+            .withVariable(INPUT_COLLECTION, List.of(1, 2, 3))
+            .create();
+
+    completeNthJob(processInstanceKey, 1);
+
+    // an incident is created because the variable `undefined_var` is missing
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    // when the missing variable is provided
+    ENGINE
+        .variables()
+        .ofScope(processInstanceKey)
+        .withDocument(Maps.of(entry("undefined_var", 1)))
+        .update();
+
+    // and we resolve the incident
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+
+    // and complete the other jobs
+    completeNthJob(processInstanceKey, 2);
+    completeNthJob(processInstanceKey, 3);
+
+    // then the process is able to complete
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_COMPLETED)
+                .withElementType(BpmnElementType.PROCESS)
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted()
+                .exists())
+        .describedAs("the process has completed")
+        .isTrue();
+
+    // and all results can be collected
+    // note, that this failed in a bug where the task was completed at the same time the incident
+    // was created. If the problem was resolved and the other tasks completed, the multi instance
+    // would still complete normally, but would not have collected the output of the first task.
+    // for more information see: https://github.com/camunda-cloud/zeebe/issues/6546
+    assertThat(
+            RecordingExporter.variableRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .withName("results")
+                .getLast())
```

Comment:
We should limit the record stream (e.g. by limiting it to the process instance). Otherwise, the test blocks 5 seconds in `.getLast()` (see [here](https://ci.zeebe.camunda.cloud/job/camunda-cloud/job/zeebe/job/korthout-6546-add-test/2/testReport/io.camunda.zeebe.engine.processing.incident/MultiInstanceIncidentTest(java-testrun)/)).